src/lib/strings/url-helpers.ts

```diff
@@ -166,3 +166,25 @@ export function getYoutubeVideoId(link: string): string | undefined {
   }
   return videoId
 }
+
+export function linkRequiresWarning(uri: string, label: string) {
+  try {
+    const urip = new URL(uri)
+    label = label.replace(/^http(s?):\/\//i, '')
+    return !label.startsWith(urip.hostname)
```

Comment:
URL#host and URL#hostname are normalized, so `new URL('https://Example.com')` with `Example.com` wouldn't match at all

Reply:
Updated the code, it normalizes now and ignores case